pyhap/accessory_driver.py

```diff
@@ -254,20 +259,27 @@ def config_changed(self):
         """Notify the driver that the accessory's configuration has changed.
 
         Persists the accessory, so that the new configuration is available on
-        restart. Also, updates the mDNS advertisment, so that iOS clients know they need
+        restart. Also, updates the mDNS advertisement, so that iOS clients know they need
         to fetch new data.
         """
         self.persist()
-        self.update_advertisment()
+        self.update_advertisement()
 
-    def update_advertisment(self):
+    def update_advertisement(self):
         """Updates the mDNS service info for the accessory."""
-        self.advertiser.unregister_service(self.mdns_service_info)
+        # TODO: Find out what causes zeroconf errors.
+        try:
+            self.advertiser.unregister_service(self.mdns_service_info)
+        except:  # KeyError (not propagated by zeroconf)
+            logger.error(WARNING_MSG_UNREGISTER_SERVICE)
         self.mdns_service_info = AccessoryMDNSServiceInfo(self.accessory,
                                                           self.address,
                                                           self.port)
         time.sleep(0.1)  # Doing it right away can cause crashes.
-        self.advertiser.register_service(self.mdns_service_info)
+        try:
+            self.advertiser.register_service(self.mdns_service_info)
+        except:  # NonUniqueNameException (not propagated by zeroconf)
```

Comment:
why don't we except only for NonUniqueName?

Reply:
I only got: `Unknown error, possibly begin NonUniqueName`. When I only tried to catch the `NonUniqueName`, it failed.

Reply:
How do you test? The NonUnique should propagate from here https://github.com/jstasiak/python-zeroconf/blob/master/zeroconf.py#L1875

Reply:
I modified the `main.py` to:

```py
def run(self):
    self.temp_char.set_value(20)
    while not self.run_sentinel.wait(0.5):
        print('==> Config Changed')
        self.config_changed()
```

That seems to reliably produce these errors and the second one isn't caught. Although that doesn't happen with only `except:` either. I change it.

Reply:
Interestingly if I add `self.run_sentinel.wait(1)` belove `set_value` I don't get these errors.